compiler/testData/codegen/box/reflection/properties/propertyName.kt

```diff
@@ -0,0 +1,12 @@
+class A {
+    val a = ""
+    val b = ""
+}
+
+fun box(): String {
+    val a = A::a.name
+    if (a != "a") return "Fail $a"
+    val b = A::`b`.name
+    if (b != "b") return "Fail $b"
+    return "OK"
```

Comment:
This test doesn't check that no classes were generated for the property reference. I don't think it's possible in a `box` test, please try a bytecode listing test instead (`compiler/testData/codegen/bytecodeListing`).

Also please test constructors and functions